lib/api/apiUtils/bucket/getNotificationConfiguration.js

```diff
@@ -31,6 +31,45 @@ function getNotificationConfiguration(parsedXml) {
         error = error.addMetadataEntry('invalidArguments', invalidTargets);
         return { error };
     }
+
+    // getting invalid events
+    const invalidEvents = [];
+    if (!config.hasColdStorage) {
```

Comment:
Restore/coldstorage is based on lifecycle transition; so these events should be rejected if transitions are not supported by the product.
(I don't think we have different flags for hot vs cold transition)